### DISCUSSION TITLE
Set custom User attributes as "safe"

### DISCUSSION THREAD

**Discussion by matthieugouel** [fastapi-users/fastapi-users#819] [Q&A] [answered]
Hi 👋 !

I just discover your project and it's neat! Thanks for the hard work! 

I have a quick question, it seams that we cannot set a custom attribute of the user model as "safe", i.e., prevent the user to patch it.  

It could be nice to do that if you have for instance a boolean that give the ability for a user to use a specific feature. Another example could be `is_premium_user` boolean flag that inform that the user have a premium account. 

I'm not sure on how to implement this kind of behaviour as is, would you have an advice ? I was considering just prevent the user to do patch at all but it's a little overkill. 

Best, 
Matthieu.

**Comment by frankie567** [comment, marked as answer]
Salut @matthieugouel!

Actually, this behavior is controlled by two methods on the user model: `create_update_dict` ("safe") and `create_update_dict_superuser` ("not safe").

https://github.com/fastapi-users/fastapi-users/blob/c759bb691569306336e099a976aa51e352322143/fastapi_users/models.py#L7-L21

To achieve what you want, just overload those methods with the fields you want to protect. Since you'll likely need it both for create and update, my advice would be to put them in a separate class ("a mixin"), and make your user models inherit from it:

```py
class CustomCreateUpdateDictModel(BaseModel):
    def create_update_dict(self):
        return self.dict(
            exclude_unset=True,
            exclude={
                "id",
                "is_superuser",
                "is_active",
                "is_verified",
                "oauth_accounts",
                "is_premium",  # Your fields here
            },
        )


class UserCreate(CustomCreateUpdateDictModel, models.BaseUserCreate):
    pass


class UserUpdate(CustomCreateUpdateDictModel, models.BaseUserUpdate):
    pass
```

Notice the order of inheritance: your custom mixin should appear first so it correctly overloads the original method.

**Reply by matthieugouel** [reply]
Marvellous! Merci !! 

Have a great day,
Matthieu.

**Reply by felipebustillo** [reply]
Hi! Can this properties still be editable by a superuser?
